xla/service/gpu/fusions/transforms/expand_float_ops.cc

```diff
@@ -516,9 +630,42 @@ struct RewriteExtFPattern : public mlir::OpRewritePattern<ma::ExtFOp> {
     }
 
     mlir::ImplicitLocOpBuilder b(op.getLoc(), rewriter);
-    rewriter.replaceOp(op, EmitFloatConversion(src, dst_ty, b));
+    bool use_intrinsic = allow_intrinsics && (src.getType().isFloat8E4M3FN() ||
+                                              src.getType().isFloat8E5M2());
+    rewriter.replaceOp(op, use_intrinsic
+                               ? EmitExtFromF8Intrinsic(src, dst_ty, b)
+                               : EmitFloatConversion(src, dst_ty, b));
     return mlir::success();
   }
+
+  Value EmitExtFromF8Intrinsic(Value value, mlir::FloatType to_ty,
+                               mlir::ImplicitLocOpBuilder& b) const {
+    // Extend the smaller type to the FP16 type using the intrinsic, and then
+    // to the destination type. In the case of BF16 go through the intermediate
+    // FP32 type (as there's no F2F op for f16->bf16).
+    Value input = b.create<ml::ZExtOp>(
```

Comment:
Let's add an assertion here that value.getType is f8e4m3fn or f8e5m2.

Reply:
Added.